src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowTests.cs

```diff
@@ -2034,6 +2034,386 @@ static bool Set(out int x)
+                // (9,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(9, 15),
+                // (16,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(16, 15),
+                // (17,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 3
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(17, 15));
+        }
+
+        [Fact]
+        public void CondAccess_NullCoalescing_06()
+        {
+            var source = @"
+class C
+{
+    void M1(C c1)
+    {
+        int x, y;
+        _ = c1?.M(x = y = 0) ?? true
+            ? x.ToString() // 1
+            : y.ToString();
+    }
+
+    void M2(C c1)
+    {
+        int x, y;
+        _ = c1?.M(x = y = 0) ?? false
+            ? x.ToString()
+            : y.ToString(); // 2
+    }
+
+    bool M(object obj) { return true; }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(8, 15),
+                // (17,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(17, 15));
+        }
+
+        [Fact]
+        public void CondAccess_NullCoalescing_07()
+        {
+            var source = @"
+class C
+{
+    void M1(bool b, C c1)
+    {
+        int x, y;
+        _ = c1?.M(x = y = 0) ?? b
+            ? x.ToString() // 1
+            : y.ToString(); // 2
+    }
+
+    bool M(object obj) { return true; }
+}
+";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,15): error CS0165: Use of unassigned local variable 'x'
+                //             ? x.ToString() // 1
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(8, 15),
+                // (9,15): error CS0165: Use of unassigned local variable 'y'
+                //             : y.ToString(); // 2
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(9, 15));
+        }
+
+        [Fact]
+        public void CondAccess_NullCoalescing_08()
+        {
+            var source = @"
+class C
+{
+    void M1(C c1)
+    {
+        int x, y;
+        _ = (c1?.M(x = y = 0)) ?? false
+            ? x.ToString()
+            : y.ToString(); // 1
+    }
+
+    void M2(C c1)
+    {
+        int x, y;
+        _ = c1?.M(x = y = 0)! ?? false
```

Comment:
>! [](start = 28, length = 1)

What are these suppressions for?

Reply:
This demonstrates that the analysis works when the conditional access is contained within a nullable suppression.